src/main/service/userManagementService.ts

```diff
@@ -0,0 +1,330 @@
+        const itemValuesArray = itemValues.split(',');
+        itemValuesArray.forEach(item => {
+          itemsArray.push({
+            href: queryUrl + '&clear=' + urlParam + item,
+            text: constValue[item],
+          });
+        });
+      } else {
+        itemsArray.push({
+          href: queryUrl + '&clear=' + urlParam + itemValues,
+          text: itemValues,
+        });
+      }
+      categoryObject['items'] = itemsArray;
+    }
+    return categoryObject;
+  }
+
+  /**
+   * Handles removing filters based off what is in the clear object.
+   */
+  public handleFilterClearing(body: any) {
+    if (body.clear === 'all') {
+      body = {};
+    } else {
+      const clearBody = body.clear.split('=');
+      switch (clearBody[0]) {
+        case 'email':
+          delete body.email;
+          break;
+        case 'userId':
+          delete body.userId;
+          break;
+        case 'userProvenanceId':
+          delete body.userProvenanceId;
+          break;
+        case 'roles':
+          body.roles = body.roles.split(',').filter(f => f !== clearBody[1]);
+          break;
+        case 'provenances':
+          body.provenances = body.provenances.split(',').filter(f => f !== clearBody[1]);
+          break;
+      }
+      delete body.clear;
+      delete body.page;
+    }
+    return body;
+  }
+
+  /**
+   * Build the summary list for the manage user screen.
+   */
+  public buildManageUserSummaryList(rawData: any): object {
+    const rows = [];
+    rows.push(this.buildRowItem('User ID', rawData.userId));
+    rows.push(this.buildRowItem('Email', rawData.email));
+    rows.push(this.buildRowItem('Role', formattedRoles[rawData.roles],
+      '/update-user?id=' + rawData.userId));
+    rows.push(this.buildRowItem('Provenance', formattedProvenances[rawData.userProvenance]));
+    rows.push(this.buildRowItem('Provenance ID', rawData.provenanceUserId));
+    rows.push(this.buildRowItem('Creation Date', this.formatDate(rawData.createdDate)));
+
+    if (rawData.roles != 'VERIFIED') {
+      rows.push(this.buildRowItem('Last Sign In', this.formatDate(rawData.lastSignedInDate)));
+    } else {
+      rows.push(this.buildRowItem('Last Verified', this.formatDate(rawData.lastVerifiedDate)));
+    }
+
+    return {rows};
+  }
+
+  /**
+   * Build the row item for the manage user summary list.
+   */
+  private buildRowItem(headingText: string, rowValue: string, actions: any = false) {
+    const rowObject = {
+      key: {
+        text: headingText,
+      },
+      value: {
+        text: rowValue,
+      },
+    };
+
+    if(actions && rowValue != 'Media') {
+      const items = [];
+      items.push({
+        href: actions,
+        text: 'Change',
+      });
+      rowObject['actions'] = {items};
+    }
+    return rowObject;
+  }
+
+  /**
+   * Format and return the date with the correct format.
+   */
+  private formatDate(rawDate: any) {
+    return moment(rawDate).format('DD/MM/YYYY HH:mm:ss');
```

Comment:
I am not sure the time is going to be formatted correctly during daylight saving. It will be formatted correctly after 30 October this year but if the datetime is during daylight saving, it will be one hour off. The display time should be in the local UK time regardless of whether it is GMT or BST.

In account management, we have the following line which indicates the datetime is stored in the UTC format:

LocalDateTime localDateTime = LocalDateTime.now(ZoneId.of("UTC"));

When the date time is received from Account management in front-end it lost the 'Z' suffix the UTC indicator because converting a LocalDateTime to a string does not take into account the time format during serialisation. I have this PR here fixing that for data management. Think we will need to do the same for account management timestamp.

https://github.com/hmcts/pip-data-management/pull/208/files#diff-3830eb72074ec4e3ff516b380632461e8a87e5272503a024ebcc4591d5b0cb6a

After that we need to format the time in front end using the following line and that should format the time correctly.

moment.utc(rawDate).tz('europe/london').format('DD/MM/YYYY HH:mm:ss');

Reply:
Good spot. Have updated this now and also in the account management PR :)